sensor/tests/resource/pod/pod_test.go

```diff
@@ -129,20 +130,17 @@ func (s *PodHierarchySuite) Test_ParentlessPodsAreTreatedAsDeployments() {
 
 func (s *PodHierarchySuite) Test_DeleteDeployment() {
 	s.testContext.RunBare("Removing a deployment should send empty violation", func(t *testing.T, testC *resource.TestContext, _ map[string]k8s.Object) {
-		deleteDep, err := testC.ApplyFileNoObject(context.Background(), resource.DefaultNamespace, NginxDeployment)
-		require.NoError(t, err)
-
 		var id string
+		k8sDeployment := &appsv1.Deployment{}
+		deleteDep, err := testC.ApplyFile(context.Background(), resource.DefaultNamespace, NginxDeployment, k8sDeployment)
+		require.NoError(t, err)
+		id = string(k8sDeployment.GetUID())
```

Comment:
Cool that this works! I wasn't sure that the ID would be set in the object after a applying it.

Reply:
Definitely makes things easier 🙂